src/handlers/discord/userStateHandler.js

```diff
@@ -0,0 +1,203 @@
+      updates.push({id: `user_${userIndex}_nick`, value: user.nick});
+      updates.push({
+        id: `user_${userIndex}_volume`,
+        value: Math.round(user.volume),
+      });
+      updates.push({id: `user_${userIndex}_avatar`, value: user.base64Avatar});
+      updates.push({id: `user_${userIndex}_avatarID`, value: user.user.avatar});
+
+      this.TPClient.stateUpdateMany(updates);
+
+      // Divide by 2 to convert range from 0-200 to 0-100
+      let volume = convertVolumeToPercentage(this.DG.currentVoiceUsers[data.user.id].volume) / 2;
+      this.TPClient.connectorUpdate(
+        `discord_voice_volume_action_connector|voiceUserList_connector=${userIndex}`,
+        volume
+      );
+
+      // This is for CustomVoiceActivityusers which will be defined by the end user to track specific users
+      if (this.DG.customVoiceAcivityUsers.hasOwnProperty(data.user.id)) {
+        let userIndex = this.DG.customVoiceAcivityUsers[data.user.id];
+        let updates = [
+          {
+            id: `${userIndex}_id`,
+            value: this.DG.currentVoiceUsers[data.user.id].user.id,
+          },
+          {
+            id: `${userIndex}_nick`,
+            value: this.DG.currentVoiceUsers[data.user.id].nick,
+          },
+          {
+            id: `${userIndex}_avatar`,
+            value: this.DG.currentVoiceUsers[data.user.id].user.base64Avatar,
+          },
+          {
+            id: `${userIndex}_avatarID`,
+            value: this.DG.currentVoiceUsers[data.user.id].user.avatar,
+          },
+          {
+            id: `${userIndex}_mute`,
+            value: this.DG.currentVoiceUsers[data.user.id].mute ? "On" : "Off",
+          },
+          {
+            id: `${userIndex}_deaf`,
+            value: this.DG.currentVoiceUsers[data.user.id].voice_state.deaf ? "On" : "Off",
+          },
+          {
+            id: `${userIndex}_self_deaf`,
+            value: this.DG.currentVoiceUsers[data.user.id].voice_state.self_deaf ? "On" : "Off",
+          },
+          {
+            id: `${userIndex}_self_mute`,
+            value: this.DG.currentVoiceUsers[data.user.id].voice_state.self_mute ? "On" : "Off",
+          },
+          {
+            id: `${userIndex}_server_mute`,
+            value: this.DG.currentVoiceUsers[data.user.id].voice_state.mute ? "On" : "Off",
+          },
+          {
+            id: `${userIndex}_volume`,
+            value: Math.round(this.DG.currentVoiceUsers[data.user.id].volume),
+          },
+        ];
+        this.TPClient.stateUpdateMany(updates);
+      }
+    } catch (error) {
+      logIt("ERROR", `updateUserStates: ${error}`);
+    }
+  };
+
+  deleteUserStates = async (data) => {
+    if (this.DG.Client.user.id === data.user.id) {
+      logIt("INFO", "Client User has left the voice channel");
+      await this.clearUserStates();
+    } else {
+      logIt("INFO", `${data.nick} (${data.user.id}) has left the voice channel`);
```

Comment:
Change to DEBUG as this could be alot of noise in the logs
```suggestion
      logIt("DEBUG", "Client User has left the voice channel");
      await this.clearUserStates();
    } else {
      logIt("DEBUG", `${data.nick} (${data.user.id}) has left the voice channel`);
```